pkg/controller/job/job_controller.go

```diff
@@ -521,7 +471,7 @@ func (jm *JobController) patchJobResource(j *batch.Job, pods []*v1.Pod) error {
 			continue
 		}
 
```

Comment:
I should have seen this earlier but please gate the changes in lines 298, 368, 563 which calls enqueueResourceUpdateForRetry in the feature gate.

Reply:
nice catch. fixed fro both job and deployment controller